tests/test_binning.py

```diff
@@ -1,44 +1,72 @@
 import numpy as np
+import pandas as pd
 import pytest
 
-from sed.binning import _hist_from_bin_ranges, _hist_from_bins
-
-sample1d = np.random.randn(int(1e2), 1)
-sample2d = np.random.randn(int(1e2), 2)
-sample3d = np.random.randn(int(1e2), 3)
-bins1d = (95,)
-bins2d = (95, 34)
-bins3d = (95, 34, 27)
-ranges1d = np.array([[1, 2]])
-ranges2d = np.array([[1, 2], [1, 2]])
-ranges3d = np.array([[1, 2], [1, 2], [1, 2]])
-arrays1d = np.linspace(*ranges1d[0],bins1d[0])
-arrays2d = [np.linspace(*ranges2d[i],bins2d[i]) for i in range(2)]
-arrays3d = [np.linspace(*ranges3d[i],bins3d[i]) for i in range(3)]
+from .helpers import get_linear_bin_edges
+from sed.binning import _hist_from_bin_range
+from sed.binning import numba_histogramdd
+
+
+sample = np.random.randn(int(1e2), 3)
+columns = ["x", "y", "z"]
+sample_df = pd.DataFrame(sample, columns=columns)
+bins = tuple(np.random.randint(5, 50, size=3))
+ranges = 0.5 + np.random.rand(6).reshape(3, 2)
+ranges[:, 0] = -ranges[:, 0]
+ranges = tuple(tuple(r) for r in ranges)
+
+arrays = [
+    get_linear_bin_edges(np.linspace(r[0], r[1], b))
+    for r, b in zip(ranges, bins)
+]
+
 
 @pytest.mark.parametrize(
     "_samples",
-    [sample1d, sample2d, sample3d],
+    [sample[:, :1], sample[:, :2], sample[:, :3]],
     ids=lambda x: f"samples:{x.shape}",
 )
 @pytest.mark.parametrize(
     "_bins",
-    [bins1d, bins2d, bins3d],
+    # [tuple(bins[:i+1]) for i in range(3)],
+    [bins[:1], bins[:2], bins[:3]],
     ids=lambda x: f"bins:{len(x)}",
 )
-def test_hist_Nd_error_is_raised(_samples, _bins):
+def test_histdd_error_is_raised(_samples, _bins):
     with pytest.raises(ValueError):
         if _samples.shape[1] == len(_bins):
             pytest.skip("Not of interest")
-        _hist_from_bin_ranges(_samples, _bins, ranges1d)
+
+        _hist_from_bin_range(_samples, _bins, ranges[0])
```

Comment:
This test fails for me, as long as numba @jit is turned on. I have no clue why. The old test function, which uses essentially the same input function does work also with numba, with no change to the code. Absolutely mysterious to me...

Reply:
That is indeed the reason. However, I don't know how to fix it using the current setup of the tests. I seem not to be able to convert this into a np.array and still keep the tests running. It always throws the error, no matter how many np.asarrays I put and where.